app/controllers/concerns/submission_scoring.rb

```diff
@@ -12,22 +12,25 @@ def collect_test_results(submission)
         output = execute_test_file(file, submission)
         assessment = assessor.assess(output)
         passed = ((assessment[:passed] == assessment[:count]) and (assessment[:score]).positive?)
-        testrun_output = passed ? nil : 'message: ' + output[:message].to_s + "\n stdout: " + output[:stdout].to_s + "\n stderr: " + output[:stderr].to_s
+        testrun_output = passed ? nil : 'status: ' + output[:status].to_s + "\n stdout: " + output[:stdout].to_s + "\n stderr: " + output[:stderr].to_s
         unless testrun_output.blank?
           submission.exercise.execution_environment.error_templates.each do |template|
             pattern = Regexp.new(template.signature).freeze
             StructuredError.create_from_template(template, testrun_output, submission) if pattern.match(testrun_output)
           end
         end
-        Testrun.new(
+        testrun = Testrun.create(
```

Comment:
Any specific reason you decided for `.new().save` instead of `.save`? With `save`, the result is only `true` or `false` whether the entry was saved or not. With `create`, the newly created testrun is returned (and can be referenced later).

Reply:
Nope, no specific reason here.